articles/container-components-in-react/index.md

```diff
@@ -0,0 +1,178 @@
+
+Let’s start by thinking about the function(s) that the search component would perform. A search component should be a able to perform basic tasks like searching and filtering.
+
+To keep things simple, let's use an array of objects as the data to search from, instead of using an API.
+
+Let's create an array of objects containing simple information about books that will be in this component:
+
+```js
+// bookshelf.js
+export const bookShelf = [
+  {
+    bookName: "Eloquent JavaScript",
+    author: "Marijn Haverbeke",
+  },
+  {
+    bookName: "CSS The Definitive Guide",
+    author: "Eric A. Meyer & Estelle Weyl",
+  },
+  {
+    bookName: "Automate the boring stuff with Python",
+    author: "Al Sweigart",
+  },
+  {
+    bookName: "Make: JavaScript Robotics",
+    author: "Donovan Buck, Bryan Hughes et al",
+  },
+];
+```
+
+The data in `authors.js` will serve as the information that would be mapped onto the `authorsPage` so the search component can be able to filter the list of authors once the user starts typing into the input field.
+
+Let's create the components needed in this app below and I'll explain how everything works.
+
+The `Search` component presents the UI (i.e. It renders an input element where users can filter the books in the list.) to the user. It makes use of React props to pass data down to other components such as the author’s page container component.
+
+```js
+// search.js
+import React from “react”
+
+const Search = ({ val, onSearch, ...props }) => {
+  return (
+    <input
+	  type="text"
+	  value={val}
+	  placeholder="Search for any book"
+	  onChange={onSearch}
+	  {...props}
+	/>
+  );
+};
+
+export default Search
+```
+
+The `Books` component performs almost the same function with the search component. Its only difference has to with the usage of React props to share data amongst the other components in the app.
+
+Take a look at how the `bookShelf` component is being imported into the authors container and how it is being used in the `Books` component.
+
+```js
+// books.js
+const Books = ({ books }) => {
+  return (
+  {books.map((bookShelf, index) => {
+    return (
+	  <ul key={index}>
+	    <li>Book: {bookShelf.bookName}</li>
+		<li>Author: {bookShelf.author}</li>
+	  </ul>
+    )
+  })}
+  )
+}
+export default Books
+```
+
+The `authorsPage` component illustrates how a container component can perform two functions, rendering the UI and housing the logic of that container.
+
+Notice how the usage of React Hooks is being used to store/hold the state of the component, which also is among the features of a container component, they are usually stateful components.
+
+The text that user types into the input field is being stored in an array so that we’d be able to perform a filtering functionality using [JavaScript’s filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/) method.
```

Comment:
The `e.target.value` will provide a string from the input, not an array. You're setting the initial state of searchTerm as an empty string instead of an empty array.

Reply:
Wow!

How did I not notice this 😢

Reply:
Hi @Caleb335. The line still mentions the text is being stored in an array. Just say it's stored in the state as a string.